### ISSUE TITLE
Modules not exported inline (export class MyModule) are not recognized by language service

### ISSUE THREAD

**Issue by naturalfreak** [angular/angular#42184]
os - Ubuntu 20.04
nodejs - 16.1.0

1. Download problem.zip
2. npm install
3. Install ext. @angular/language-service for VSCode
4. Open project with VSCode

latest  version of extention - @angular/language-service for VSCode
[problem.zip](https://github.com/angular/vscode-ng-language-service/files/6492040/problem.zip) - project

When using this configuration I got errors:

in src/app/dashboard/dashboard.component.html

```
'drawer-component' is not a known element:
1. If 'drawer-component' is an Angular component, then verify that it is part of this module.
2. If 'drawer-component' is a Web Component then add 'CUSTOM_ELEMENTS_SCHEMA' to the '@NgModule.schemas' of this component to suppress this message.ngtsc(-998001)
```
in src/shared/drawer-component/index.ts

```
Appears in the NgModule.imports of AppModule, but could not be resolved to an NgModule class.

Is it missing an @NgModule annotation?(-996002)

```
Settings:

![Screenshot from 2021-05-17 10-43-43](https://user-images.githubusercontent.com/11893674/118450396-d1a1fc80-b6fc-11eb-8df5-0d193d08fa8b.png)
![Screenshot from 2021-05-17 10-43-49](https://user-images.githubusercontent.com/11893674/118450397-d2d32980-b6fc-11eb-8c41-ee45b5d3fda6.png)

Problem:
![Screenshot from 2021-05-17 10-39-23](https://user-images.githubusercontent.com/11893674/118449963-5b9d9580-b6fc-11eb-84d2-0edc4f883243.png)
![Screenshot from 2021-05-17 10-39-27](https://user-images.githubusercontent.com/11893674/118449968-5ccec280-b6fc-11eb-9100-01380143a8fa.png)

**Comment by alxhub** [comment]
Transferring this to angular/angular as this is fundamentally a compiler bug (we need to be smarter about detecting "exported" components).

**Comment by unnhao** [comment]
I have same issue, I don't know what's going on...

**Comment by atscott** [comment]
This will be addressed by #42184. It _might_ miss the release this week but will definitely be in by next week.
